### PR TITLE
Mimes now break oath via spell again, not emoting

### LINKED ISSUE #4747: (WEB REPORT BY: orangerush REMOTE: 198.245.63.50:7777) Mime breaks his vows (?)
ec3073efc7759801daa51b26c703bb4d14fda35b 

Mime no longer has the option to break his vow and speak. Only dying and being cloned allows you to speakk

### PR DESCRIPTION
**Pull request by Intigracy**
tested-stamp

fixes #4747

### CODE REVIEW

**Comment by Intigracy** [comment]
unintuitive as fuck

File: code/game/jobs/job/civilian.dm [hunk 1]
```diff
@@ -309,10 +309,40 @@
...
+	for(var/mob/living/carbon/human/M in targets)
+		var/response = alert(M, "Are you -sure- you want to break your oath of silence?\n(This removes your ability to create invisible walls and cannot be undone!)","Are you sure you want to break your oath?","Yes","No")
+		if(response != "Yes")	return
+		M.miming=0
+		for(var/spell/aoe_turf/conjure/forcewall/mime/spell in M.spell_list)
+			M.remove_spell(spell)
+			qdel(spell)
+		for(var/spell/targeted/oathbreak/spell in M.spell_list)
+			M.remove_spell(spell)
+			qdel(spell)
```
Comment (clusterfack) [inline code/game/jobs/job/civilian.dm]: Is qdel necessary here?

Comment (Intigracy) [inline code/game/jobs/job/civilian.dm]: I haven't checked remove_spell for whether or not it qdels, maybe not.

**Comment by Intigracy** [comment]
```
/obj/screen/movable/spell_master/proc/remove_spell(var/spell/spell)
    returnToPool(spell.connected_button)

    spell.connected_button = null

    if(spell_objects.len)
        toggle_open(showing + 1)
    else
        returnToPool(src)
```

```
/mob/proc/remove_spell(var/spell/spell_to_remove)
    if(!spell_to_remove || !istype(spell_to_remove))
        return

    if(!(spell_to_remove in spell_list))
        return

    if(!spell_masters || !spell_masters.len)
        return

    if(mind && mind.wizard_spells)
        mind.wizard_spells.Remove(spell_to_remove)
    spell_list.Remove(spell_to_remove)
    for(var/obj/screen/movable/spell_master/spell_master in spell_masters)
        spell_master.remove_spell(spell_to_remove)
    return 1
```

**Comment by clusterfack** [comment]
Yeah I'm just not sure remove_spell works to be honest. It was never finished and only lightly used for a Destroy() proc to my knowledge.

I think I touched it up but I'm not sure it works in the way you are trying to use it so use so make sure.

**Comment by Intigracy** [comment]
Well I tested the code as-is and it functions correctly, if that's what you mean. It removes the spells.

Unless you mean I don't need a line somewhere.

**Comment by clusterfack** [comment]
Yeah the qdel is totally unnecessary in both cases

**Comment by ihadtoregisterforthis** [comment]
This will lead to a lot of accidents, even with confirmation needed.

**Comment by Intigracy** [comment]
How?

**Comment by Intigracy** [comment]
I could alternatively make it an input box that makes them type "I'm a bad mime." to break the vow.

**Comment by ihadtoregisterforthis** [comment]
It's a big button on your screen, for starters

**Comment by Intigracy** [comment]
That you have to click and then click yes on the popup.

If you can't even read the warning you deserve not to have mime powers.

**Comment by Intigracy** [comment]
Did what you said, @clusterfack 

Tested, you were right.

**Comment by ihadtoregisterforthis** [comment]
If you are going to break your oath you don't deserve being a mime anyways.

**Pull request merged**